File: project-finicalHystrix/src/main/java/com/finicalHystrix/T.scala
package com.finicalHystrix

import org.apache.flink.streaming.api.TimeCharacteristic
import org.apache.flink.streaming.api.scala._
import org.apache.flink.table.api.EnvironmentSettings
import org.apache.flink.table.api.bridge.scala._
import org.apache.flink.types.Row

object T {
  def main(args: Array[String]): Unit = {
    val env = StreamExecutionEnvironment.getExecutionEnvironment
    env.setParallelism(1)
    env.setStreamTimeCharacteristic(TimeCharacteristic.ProcessingTime)
    val settings = EnvironmentSettings.newInstance().useBlinkPlanner().inStreamingMode().build()
    val tableEnv = StreamTableEnvironment.create(env, settings)

    val kuduMasters = "real-time-006"
    val sql =
      """
        |create table finicalinfo(
        |relation_id	string,
        |event_type	string,
        |user_id	string,
        |amt float,
        |event_dt	bigint,
        |should_repay_time	bigint,
        |repay_state	string,
        |actual_repay_time	bigint
        |)with(
        |'connector.type'='kudu',
        |'kudu.table'='qinghua_finicalinfo',
        |'kudu.masters'='real-time-006'
        |)
        |""".stripMargin

    tableEnv.executeSql(sql)
    val table = tableEnv.sqlQuery("select count(1) from finicalinfo")

    env.execute("aaa")
  }
}

File: db-kudu2/src/main/java/org/apache/flink/connectors/kudu/test/T.java
package org.apache.flink.connectors.kudu.test;

import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.table.api.EnvironmentSettings;
import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;

public class T {
    public static void main(String[] args) {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        EnvironmentSettings settings = EnvironmentSettings.newInstance().useBlinkPlanner().inStreamingMode().build();
        StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env, settings);
        String sql = "create table finicalinfo(\n" +
                "relation_id\tstring,\n" +
                "event_type\tstring,\n" +
                "user_id\tstring,\n" +
                "amt float,\n" +
                "event_dt\tbigint,\n" +
                "should_repay_time\tbigint,\n" +
                "repay_state\tstring,\n" +
                "actual_repay_time\tbigint\n" +
                ")with(\n" +
                "'connector'='kudu',\n" +
                "'kudu.table'='qinghua_finicalinfo'\n" +
                ")";
        tableEnv.executeSql(sql);
        tableEnv.executeSql("insert into finicalinfo values('005','0','user1',2.0,1612406576,1612407176,'0',0)");
    }
}
